src/vsphere_cpi/spec/unit/cloud/vsphere/storage_picker_spec.rb

```diff
@@ -71,47 +71,141 @@ module VSphereCloud
+        allow(vm_type).to receive(:storage_policy_datastores).with(vm_type_storage_policy).and_return(
+            compatible_datastores_vm_type)
+        allow(vm_type).to receive_message_chain(:datacenter, :ephemeral_pattern).and_return(global_ephemeral_pattern)
+        allow(global_config).to receive(:vm_storage_policy_name).and_return(global_storage_policy)
       end
 
-      context 'with datastore clusters' do
-        context 'and sdrs enabled on some' do
-          let(:datastore_clusters) { [sdrs_enabled_datastore_cluster1, sdrs_enabled_datastore_cluster2, sdrs_disabled_datastore_cluster] }
-          context 'along with datastores' do
-            let(:datastore_names) { ['ds-1', 'ds-2'] }
-            it 'includes a pattern constructed from datastores and datastores from all sdrs enabled datastore cluster' do
-              expect(subject).to eq('^(ds\-1|ds\-2|sp\-1\-ds\-1|sp\-2\-ds\-1)$')
-            end
+      context 'with storage policy defined in vm_type' do
+        context 'with datastore pattern(vm_type, global both) defined and global storage policy is also defined' do
+          let(:global_ephemeral_pattern) { 'global-ds-0' }
+          let(:datastore_clusters) { ['fake-1'] }
+          let(:datastore_names) { ['ds-0', 'ds-1'] }
+          let(:vm_type_storage_policy) {'Gold Policy'}
+          let(:compatible_datastores_vm_type) { [ double('Datastore', name: 'ds-5'), double('Datastore', name: 'ds-6') ]}
+          let(:compatible_pattern_vm_type) { '^(ds\-5|ds\-6)$' }
+          let(:global_compatible_datastores) { [ double('Datastore', name: 'ds-7'), double('Datastore', name: 'ds-8') ]}
+          let(:global_compatible_datastores_pattern) { '^(ds\-7|ds\-8)$' }
+          let(:global_storage_policy) { 'Global Gold Policy' }
+
+          it 'includes all the compatible datastores from vm_type storage policy' do
+            expect(vm_type).to_not receive(:datastore_names)
+            expect(vm_type).to_not receive(:datastore_clusters)
+            expect(vm_type).to_not receive(:datacenter)
+            expect(global_config).to_not receive(:vm_storage_policy_name)
+            expect(vm_type).to receive(:storage_policy_datastores).with(
+                vm_type_storage_policy).and_return(compatible_datastores_vm_type)
+            expect(vm_type).to receive(:storage_policy_name).exactly(4).times
+            expect(subject).to eq([compatible_pattern_vm_type, vm_type_storage_policy])
+
           end
+        end
+      end
 
-          context 'and no datastores' do
-            let(:datastore_names) { [] }
-            it 'includes the datastores from all sdrs enabled datastore cluster' do
-              expect(subject).to eq('^(sp\-1\-ds\-1|sp\-2\-ds\-1)$')
+      context 'with NO storage policy defined in vm_type' do
+        let(:vm_type_storage_policy) { nil }
+        context 'with datastore entities/pattern(vm_type, global both) defined and global storage policy is also defined' do
```

Comment:
nitpick  'when datastore entities/pattern(vm_type, global both) and global storage policy are defined'

Reply:
The whole spec file is defined with `with` rather than `when` so letting it be.